File: trunk/speed/median_opencv.py
#!/usr/bin/env python

"""
Load, Filter then Show an image with opencv in python
Brian Thorne 2009
"""

from VideoCapturePlayer import VideoCapturePlayer as VCP
from opencv import cv

def medianBlur(image, filterSize=43):
    """Blur an image with a particular strength filter."""
    cv.cvSmooth(image, image, cv.CV_MEDIAN, filterSize)
    return image

if __name__ == "__main__":
    title = "Median Filtered Output"
    VCP(medianBlur, title).main()

File: trunk/speed/median_opencv.cxx
/*
 *      blur_opencv.cxx
 *      
 *      Copyright 2009 Brian Thorne <brian.thorne@canterbury.ac.nz>
 * 
 *      Compile with:
 * 
 * g++ -O3 -Wall `pkg-config --cflags opencv` `pkg-config --libs opencv` -o median median_opencv.cxx videoCapturePlayer.cxx
 *  
 */


#include <iostream>
#include "videoCapturePlayer.h"

CvMat * medianBlur(CvMat *x)
{
    int filterSize = 43;
    cvSmooth(x, x, CV_MEDIAN, filterSize);
    return x;
}

int main( int argc, char** argv )
{
    std::cout << "Starting Median Blur C++ demo" << std::endl;
    std::cout << "Press 'q' to exit the program" << std::endl;
    VideoCapturePlayer vcp = VideoCapturePlayer(&medianBlur);
    vcp.init(); vcp.main();
    return 0;
}

